File: ruby/lib/steam/packets/rcon/rcon_goldsrc_request.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2008-2011, Sebastian Staudt

require 'steam/packets/steam_packet'

# This packet class represents a RCON request packet sent to a GoldSrc server
#
# It is used to request a command execution on the server.
#
# @author Sebastian Staudt
# @see GoldSrcServer#rcon_exec
class RCONGoldSrcRequest

  include SteamPacket

  # Creates a request for the given request string
  #
  # The request string has the form `rcon {challenge number} {RCON password}
  # {command}`.
  #
  # @param [String] request The request string to send to the server
  def initialize(request)
    super 0x00, request
  end

  # Returns the raw data representing this packet
  #
  # @return [String] A string containing the raw data of this request packet
  def to_s
    [0xFFFFFFFF, @content_data.string].pack('Va*')
  end

end

File: php/lib/steam/packets/rcon/RCONGoldSrcRequest.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2008-2009, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage RCONGoldSrcRequest
 */

require_once STEAM_CONDENSER_PATH . 'steam/packets/SteamPacket.php';

class RCONGoldSrcRequest extends SteamPacket
{
	public function __construct($request)
	{
		parent::__construct(0x00, $request);
	}

	public function __toString()
	{
		return pack("Va*", 0xFFFFFFFF, $this->contentData->_array());
	}
}
?>
